articles/ros_node_idl.md

```diff
@@ -70,8 +70,20 @@ If ROS 2 provided a way for upstream package authors to specify the interface re
 ### Other motivations
 
 Outside of security, there are several fascinating possibilities unlocked by having such an interface.
+
 For example, consider how this could impact [ROS 2 launch][launch_ros].
-It would be able to statically (i.e. before running anything) determine if parameter names or remappings are incorrect, among other similar sanity checks.
+Benefiting from the declared interface(s), it would be able to execute many kind of static asserts (i.e. at launch-time, before running anything) upon the whole system to be launched.
+Such assertions are e.g.:
+- Check for duplicates.
+- Check for multiple publishers on a single topic.
+- Check for message type mismatch.
+- Check for qos mismatch.
+- Check for orphan connections (i.e. a listener is connected to a topic with no publisher).
+- Determine if remappings are incorrect.
+- Determine if parameter names are incorrect.
+
+These assertions results would then be summarized in a logging file for later debugging.
```

Comment:
I'd like to see them as warnings as well. Is it worth specifying this here anyway, though? Perhaps this line should be removed.

Reply:
It is simply to mention that they can be parsed afterward for debugging, effectively producing some kind of a report. How/where is implementation details. It is indeed not bringing much to the table.